src/engines/engine-decorator-manager.js

```diff
@@ -0,0 +1,23 @@
+// @flow
+
+/**
+ * Engine decorator manager for plugins.
+ * @class EngineDecoratorManager
+ */
+class EngineDecoratorManager {
+  _decoratorGenerators: Array<EngineDecoratorGenerator> = [];
+
+  addDecorator(decoratorGenerator: EngineDecoratorGenerator): void {
+    this._decoratorGenerators.push(decoratorGenerator);
```

Comment:
add check to see if it is already provided, and with above addition of name it is easier then ever